src/openloco/windows/map.cpp

```diff
@@ -1,19 +1,1770 @@
+            townPos.y += townPos.x;
+            townPos.x = -townPos.x;
+            townPos.x += yCopy + 376;
+            townPos.y -= 8;
+
+            stringmgr::format_string(_stringFormatBuffer, town.name);
+            _currentFontSpriteBase = font::small;
+
+            auto strWidth = gfx::getStringWidth(_stringFormatBuffer);
+
+            strWidth /= 2;
+
+            townPos.x -= strWidth;
+            townPos.y -= 3;
+
+            _currentFontSpriteBase = font::small;
+            gfx::draw_string(dpi, townPos.x, townPos.y, colour::outline(colour::bright_purple), _stringFormatBuffer);
+        }
+    }
+
+    // 0x0046B806
+    static void drawScroll(window* self, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+    {
+        if (!(_dword_525E28 & (1 << 0)))
+            return;
+
+        gfx::clear_single(*dpi, palette_index::index_0A);
+
+        auto element = gfx::get_g1element(0);
+        auto offset = *_dword_F253A8;
+
+        if (mapFrameNumber & (1 << 2))
+            offset += 0x90000;
+
+        gfx::get_g1element(0)->offset = offset;
+        gfx::get_g1element(0)->width = 0x300;
+        gfx::get_g1element(0)->height = 0x300;
+        gfx::get_g1element(0)->x_offset = -8;
+        gfx::get_g1element(0)->y_offset = -8;
+        gfx::get_g1element(0)->flags = 0;
+
+        gfx::draw_image(dpi, 0, 0, 0);
+
+        gfx::get_g1element(0)->offset = element->offset;
+        gfx::get_g1element(0)->width = element->width;
+        gfx::get_g1element(0)->height = element->height;
+        gfx::get_g1element(0)->x_offset = element->x_offset;
+        gfx::get_g1element(0)->y_offset = element->y_offset;
+        gfx::get_g1element(0)->flags = element->flags;
+        gfx::get_g1element(0)->unused = element->unused;
+
+        drawVehiclesOnMap(dpi, self->current_tab + widx::tabOverall);
+
+        drawViewportPosition(dpi);
+
+        if (self->saved_view.mapX & (1 << 0))
+        {
+            drawTownNames(dpi);
+        }
+    }
+
+    static void initEvents()
+    {
+        events.on_close = onClose;
+        events.on_mouse_up = onMouseUp;
+        events.on_resize = onResize;
+        events.on_update = onUpdate;
+        events.get_scroll_size = getScrollSize;
+        events.scroll_mouse_down = event18;
+        events.event_18 = (uint32_t)0x0046B97C;
+        events.tooltip = tooltip;
+        events.prepare_draw = prepareDraw;
+        events.draw = draw;
+        events.draw_scroll = drawScroll;
+    }
+
+    static void sub_46CFF0()
+    {
+        registers regs;
+        call(0x0046CFF0, regs);
+    }
+    static void sub_46CED0()
+    {
+        registers regs;
+        call(0x0046CED0, regs);
+    }
+
     // 0x0046B490
     void open()
     {
-        call(0x0046B490);
+        auto window = WindowManager::bringToFront(WindowType::map, 0);
+
+        if (window != nullptr)
+            return;
+
+        addr<0x00113E87C, int32_t>() = 3;
+        auto ptr = malloc(0x120000);
```

Comment:
0x120000 is going to represent the map size * 2. The *2 I'm not sure why but suspect its double buffering related.